src/Task1/Vote.js

```diff
@@ -1,58 +1,104 @@
+        { name: "Python", vote: 0 },
+        { name: "Java", vote: 0 },
+        { name: "C++", vote: 0 },
+        { name: "Kotlin", vote: 0 },
+      ]
     }
-    setState(state) {
-        window.localStorage.setItem('state', JSON.stringify(state));
-        super.setState(state);
-      }
-    reset(state){
-       window.localStorage.clear(state);
-       console.log("hello");
-    //    console.log(window.localStorage.getItem('state'));
-      }     
-    vote(i){
-        let newLanguageVote=this.state.languages;
-        console.log(newLanguageVote[i].name);
-        newLanguageVote[i].vote++;
+  }
 
-        console.log("click:",i);
-        console.log(newLanguageVote[i].vote);
-        this.setState({
-            languages:newLanguageVote
-        });
+  //setting localstorage using componentDidMount and ComponentDidUpdate
+  componentDidMount() {
+    if (window.localStorage.getItem('state')) {
+      this.setState({
+        languages: JSON.parse(window.localStorage.getItem('state'))
+      })
     }
-    render() {
-        return (
-        <div>
-            <h1>Vote your languages</h1>
-            <button onClick={this.reset}>Clear All</button>
-            <div className="languages">
-            {
-                this.state.languages.map((item,i)=>
-                <div key={i} className='item'>
-                    <div className='voteCount'>{item.vote}</div>
-                    <div className='lnaguageName'>{item.name}</div>
-                    <button onClick={this.vote.bind(this,i)}>Vote ME</button>
-                </div>
-                )
-            }
-            </div>
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    window.localStorage.setItem('state', JSON.stringify(this.state.languages))
+  }
+
+ //this function do not update  when Vote=4
+  shouldComponentUpdate(prevProps, prevState) {
+    const Vote = this.state.languages.map(({ vote }) => vote)
+    for (let i = 0; i < Vote.length; i++) {
+      if (Vote[i] === 4) {
+        return false
+      }
+    }
+    return true
+  }
+
+  //Just used this function for understading this hides the whole UI 
+  componentWillUnmount(){
+    console.log("component ie hidden now");
+  }
+  resetState(state) {
+    window.localStorage.clear(state);
+    this.setState({
+      languages: [
+        { name: "Javascript", vote: 0 },
+        { name: "Python", vote: 0 },
+        { name: "Java", vote: 0 },
+        { name: "C++", vote: 0 },
+        { name: "Kotlin", vote: 0 },
+      ]
+    })
+  }
+  voteIncrement(index) {
+    let newLanguageVote = this.state.languages;
+    newLanguageVote[index].vote++;
+    this.setState({
+      languages: newLanguageVote.sort((a, b) => b.vote - a.vote)
+    });
+  }
+  render() {
+    return (
+      <div>
+        <h1>Vote your languages</h1>
+        <button onClick={this.resetState.bind(this)}>Clear All</button>
+        <div className="languages">
+          {
+            this.state.languages.map((item, index) =>
```

Comment:
Destructure the item here and then use it in the JSX: 
const {vote, name} = item;

@Poorvi-3pillar